File: 문자열/백준_1475_방번호/방번호.java
import java.util.*;

public class Main{

    public static void main(String[] args){
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int[] arr = new int[10];
        while(n >= 1){
            int i = n % 10;
            if(i == 9) i = 6;
            arr[i]++;
            n /= 10;
        }
        arr[6] = (arr[6] % 2 == 0 ) ? arr[6] / 2 : arr[6] / 2 + 1;
        int max = 0;
        for(int i : arr) if(i > max) max = i;
        System.out.println(max);

    }
}

File: 문자열/백준_1475_방번호/방번호.kt
import java.util.*

fun main(args: Array<String>) {
    var s = Scanner(System.`in`)
    var n = s.nextInt()
    var arr = IntArray(10, {0})
    while(n >= 1){
        var i = n % 10
        if(i == 9) i = 6
        arr[i]+=1
        n /= 10
    }
    if(arr[6] % 2 == 0) arr[6] /= 2
    else arr[6] = arr[6] / 2 + 1
    var max = 0
    for(i in 0 until arr.size)
        if(arr[i] > max) max = arr[i]
    println(max)
}